src/ServerTelemetryChannel/ServerTelemetryChannel.cs

```diff
@@ -304,6 +304,8 @@ public void Initialize(TelemetryConfiguration configuration)
         {
             this.Transmitter.Initialize();
 
+            this.EndpointAddress = configuration.Endpoint.Ingestion.AbsoluteUri + "v2/track";
```

Comment:
>AbsoluteUri [](start = 68, length = 11)

Are you guaranteed a trailing slash?

Reply:
Here is a suggestion. Maybe I'm being too careful but since this is SDK code I'm reviewing to be cautious.

```
if (Uri.TryCreate(configuration.Endpoint.Ingestion, "v2/track", out Uri endpointAddress) {
  this.EndpointAddress = endpointAddress.AbsoluteUri;
}
else {
..
}
```

---
In reply to: [313673505](https://github.com/microsoft/ApplicationInsights-dotnet/pull/1187#discussion_r313673505) [](ancestors = 313673505)

Reply:
In my testing, it appears that `(new Uri(someString)).AbsoluteUri` always appends the trailing slash to the input string. I can do some more testing on specifically this case to confirm. Thanks :)

Reply:
> For hierarchical URIs, if the host is not terminated with a forward slash (/), one is added.

https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8

Reply:
I wrote a boat load of tests to validate the expected behavior.
I think here it'll be safest to create a new Uri and then save the resulting AbsoluteUri. 
The Uri class will give us validation.

```
    [TestClass]
    public class UriTests
    {
        [TestMethod]
        public void TestMethod1()
        {
            var expected = "https://dc.services.visualstudio.com/";

            var test1 = new Uri("https://dc.services.visualstudio.com");
            var test2 = new Uri("https://dc.services.visualstudio.com/");

            Assert.AreEqual(expected, test1.AbsoluteUri);
            Assert.AreEqual(expected, test2.AbsoluteUri);
        }

        [TestMethod]
        public void TestMethod2()
        {
            var expected = "https://dc.services.visualstudio.com/v2/track";

            var host = new Uri("https://dc.services.visualstudio.com");

            var part1 = new Uri("v2/track", UriKind.Relative);
            var part2 = new Uri("/v2/track", UriKind.Relative);

            var test1 = new Uri(host, part1);
            var test2 = new Uri(host, part2);

            Assert.AreEqual(expected, test1.AbsoluteUri);
            Assert.AreEqual(expected, test2.AbsoluteUri);
        }

        [TestMethod]
        public void TestMethod3()
        {
            var expected = "https://dc.services.visualstudio.com/v2/track";

            var host = new Uri("https://dc.services.visualstudio.com");

            var test1 = new Uri(host, "v2/track");
            var test2 = new Uri(host, "/v2/track");

            Assert.AreEqual(expected, test1.AbsoluteUri);
            Assert.AreEqual(expected, test2.AbsoluteUri);
        }
    }
```